orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfo.groovy

```diff
@@ -72,9 +72,21 @@ class PackageInfo {
   @CompileDynamic
   @VisibleForTesting
   private Map createAugmentedRequest(Map trigger, Map buildInfo, Map request) {
+
+    if (isUrl(request.package) || !trigger) {
+      return request
+    }
+
     List<Map> triggerArtifacts = trigger.buildInfo?.artifacts ?: trigger.parentExecution?.trigger?.buildInfo?.artifacts
-    List<Map> buildArtifacts = buildInfo.artifacts
-    if ((!triggerArtifacts && !buildArtifacts) || isUrl(request.package)) {
+    List<Map> buildArtifacts = buildInfo?.artifacts
+
+    if (!buildInfo || (buildInfo && !buildArtifacts)) {
+      if (!triggerArtifacts && (trigger.buildInfo != null || trigger.parentExecution?.trigger?.buildInfo != null)) {
+        throw new IllegalStateException("Could not find artifacts in Jenkins jobs")
```

Comment:
Have we verified if this error message is displayed in `deck`?

Given it's probably something we want to show users, might be better worded as: "Jenkins trigger detected but no artifacts found. Please ensure your job is archive your x,y,z."(?) or something that might be a little more self-explanatory for people.

Otherwise preventing this behavior seems safer. I've got no idea how many people this will affect in reality.